gazebo_ros/src/gazebo_ros_properties.cpp

```diff
@@ -0,0 +1,448 @@
+}
+
+void GazeboRosPropertiesPrivate::GetLightProperties(
+  gazebo_msgs::srv::GetLightProperties::Request::SharedPtr _req,
+  gazebo_msgs::srv::GetLightProperties::Response::SharedPtr _res)
+{
+  gazebo::physics::LightPtr phy_light = world_->LightByName(_req->light_name);
+  if (phy_light == NULL) {
+      _res->success = false;
+      _res->status_message = "getLightProperties: Requested light " + _req->light_name
+       + " not found!";
+  } else {
+    gazebo::msgs::Light light;
+    phy_light->FillMsg(light);
+
+    _res->diffuse.r = light.diffuse().r();
+    _res->diffuse.g = light.diffuse().g();
+    _res->diffuse.b = light.diffuse().b();
+    _res->diffuse.a = light.diffuse().a();
+
+    _res->attenuation_constant = light.attenuation_constant();
+    _res->attenuation_linear = light.attenuation_linear();
+    _res->attenuation_quadratic = light.attenuation_quadratic();
+
+    _res->success = true;
+  }
+}
+
+void GazeboRosPropertiesPrivate::SetJointProperties(
+  gazebo_msgs::srv::SetJointProperties::Request::SharedPtr _req,
+  gazebo_msgs::srv::SetJointProperties::Response::SharedPtr _res)
+{
+  /// @todo: current settings only allows for setting of 1DOF joints
+  /// (e.g. HingeJoint and SliderJoint) correctly.
+  gazebo::physics::JointPtr joint;
+  for (unsigned int i = 0; i < world_->ModelCount(); ++i) {
+    joint = world_->ModelByIndex(i)->GetJoint(_req->joint_name);
+    if (joint) {break;}
+  }
+
+  if (!joint) {
+    _res->success = false;
+    _res->status_message = "SetJointProperties: joint not found";
+  }else{
+    for(unsigned int i = 0;i< _req->ode_joint_config.damping.size();i++)
+      joint->SetDamping(i, _req->ode_joint_config.damping[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.hi_stop.size();i++)
+      joint->SetParam("hi_stop", i, _req->ode_joint_config.hi_stop[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.lo_stop.size();i++)
+      joint->SetParam("lo_stop", i, _req->ode_joint_config.lo_stop[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.erp.size();i++)
+      joint->SetParam("erp", i, _req->ode_joint_config.erp[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.cfm.size();i++)
+      joint->SetParam("cfm", i, _req->ode_joint_config.cfm[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.stop_erp.size();i++)
+      joint->SetParam("stop_erp", i, _req->ode_joint_config.stop_erp[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.stop_cfm.size();i++)
+      joint->SetParam("stop_cfm", i, _req->ode_joint_config.stop_cfm[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.fudge_factor.size();i++)
+      joint->SetParam("fudge_factor", i, _req->ode_joint_config.fudge_factor[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.fmax.size();i++)
+      joint->SetParam("fmax", i, _req->ode_joint_config.fmax[i]);
+    for(unsigned int i = 0;i< _req->ode_joint_config.vel.size();i++)
+      joint->SetParam("vel", i, _req->ode_joint_config.vel[i]);
```

Comment:
humm `vel` doesn't seem to work for me. Not an issue with this PR though, probably a problem on Gazebo.